src/Microsoft.DotNet.XHarness.CLI/Commands/WASI/Engine/WasiTestCommand.cs

```diff
@@ -0,0 +1,132 @@
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.Wasi;
+using Microsoft.DotNet.XHarness.CLI.Commands.Wasm;
+using Microsoft.DotNet.XHarness.Common;
+using Microsoft.DotNet.XHarness.Common.CLI;
+using Microsoft.DotNet.XHarness.Common.Execution;
+using Microsoft.DotNet.XHarness.Common.Logging;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.XHarness.CLI.Commands.Wasi;
+internal class WasiTestCommand : XHarnessCommand<WasiTestCommandArguments>
+{
+    private const string CommandHelp = "Executes tests on WASI using a selected engine";
+
+    protected override WasiTestCommandArguments Arguments { get; } = new();
+    protected override string CommandUsage { get; } = "wasi test [OPTIONS] -- [ENGINE OPTIONS]";
+    protected override string CommandDescription { get; } = CommandHelp;
+
+    public WasiTestCommand() : base(TargetPlatform.WASI, "test", true, new ServiceCollection(), CommandHelp)
+    {
+    }
+
+    protected override async Task<ExitCode> InvokeInternal(ILogger logger)
+    {
+        var processManager = ProcessManagerFactory.CreateProcessManager();
+
+        string engineBinary = Arguments.Engine.Value switch
+        {
+            WasmEngine.WasmTime => "wasmtime",
+            _ => throw new ArgumentException("Engine not set")
+        };
+
+        if (!string.IsNullOrEmpty(Arguments.EnginePath.Value))
+        {
+            engineBinary = Arguments.EnginePath.Value;
+            if (Path.IsPathRooted(engineBinary) && !File.Exists(engineBinary))
+                throw new ArgumentException($"Could not find wasm engine at the specified path - {engineBinary}");
+        }
+      
+        logger.LogInformation($"Using wasm engine {Arguments.Engine.Value} from path {engineBinary}");
+        await PrintVersionAsync(Arguments.Engine.Value.Value, engineBinary);
+
+        var cts = new CancellationTokenSource();
+        try
+        {
+            var engineArgs = new List<string>();
+            engineArgs.Add(Arguments.SubCommand);
+            engineArgs.Add("--dir");
+            engineArgs.Add(".");
+            engineArgs.Add(Arguments.WasmFile);
+            engineArgs.AddRange(Arguments.EngineArgs.Value);
+            engineArgs.AddRange(PassThroughArguments);
+
+            var xmlResultsFilePath = Path.Combine(Arguments.OutputDirectory, "testResults.xml");
+            File.Delete(xmlResultsFilePath);
+
+            var stdoutFilePath = Path.Combine(Arguments.OutputDirectory, "wasi-console.log");
+            File.Delete(stdoutFilePath);
+
+            var processTask = processManager.ExecuteCommandAsync(
+                engineBinary,
+                engineArgs,
+                log: new CallbackLog(m => logger.LogInformation(m)),
```

Comment:
I tried running a test with the command and found that `m` already had a newline at the end, and `LogInformation` would add another unnecessary one. So, use `logger.LogInformation(m.Trim())`.